crypto/bio/bss_sock.c

```diff
@@ -64,7 +65,12 @@ BIO *BIO_new_socket(int fd, int close_flag)
     if (ret == NULL)
         return NULL;
     BIO_set_fd(ret, fd, close_flag);
-    return ret;
+#ifndef OPENSSL_NO_KTLS
```

Comment:
push an error code, don't printf in the library.

Reply:
The printf helps with debugging. I'll add it under an SSL_DEBUG clause. OK?

The error code wasn't propagated to the caller on purpose. We do this because the socket is created successfully regardless of the return value from ktls_enable. However, future calls to ktls_* functions will fail if this call fails. This can happen when the library is compiled with ktls but the kernel is too old. As a result, the library creates the socket successfully, even though the socket option fails.